lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -630,24 +630,42 @@ void LegalizeMD::LegalizeLDIMM(IR::Instr * instr, IntConstType immed)
     }
     else
     {
-        // ARM64_WORKITEM: This needs to be understood better
+        // Since we don't know the value yet, we're going to handle it when we do
+        // This is done by having the load be from a label operand, which is later
+        // changed such that its offset is the correct value to ldimm
+
+        // The assembly generated becomes something like
+        // Label (offset:fake)
+        // MOVZ DST, Label
+        // MOVK DST, Label
+        // MOVK DST, Label
+        // MOVK DST, Label <- was the LDIMM
+
         Assert(Security::DontEncode(instr->GetSrc1()));
```

Comment:
If this is a label, you can use ADR instead of 4 moves. Also, you can assume that user mode pointers are only 48 bits and eliminate the 4th move if you can't use ADR.

Reply:
It's not a normal label, unfortunately; we set the label's offset to a particular value, and feed that value back in (it's basically an immediate that we don't yet know the value of). There's a few more notes on the ARM32 side, notably in the arm\EncoderMD.cpp:ApplyRelocs function.

Reply:
Got it. Definitely consider whether the final value can be limited to 48 or 32 bits in order to reduce the number of moves.

Reply:
Since we don't know this early enough to be able to cleanly remove instructions, we'd end up just nop-ing out some of the movs. I'm mostly worried about that making this path more complex, since we'd have to determine at encoding time which of the instructions can be omitted.